src/matrix_free/StkSimdGatheredElementData.C

```diff
@@ -0,0 +1,161 @@
+#include "matrix_free/StkSimdGatheredElementData.h"
+
+#include "Kokkos_ExecPolicy.hpp"
+#include "Kokkos_Macros.hpp"
+#include "matrix_free/PolynomialOrders.h"
+#include "matrix_free/StkSimdConnectivityMap.h"
+#include "stk_mesh/base/Field.hpp"
+#include "stk_mesh/base/FieldBase.hpp"
+#include "stk_mesh/base/Types.hpp"
+#include "stk_ngp/Ngp.hpp"
+#include "stk_simd/Simd.hpp"
+
+namespace sierra {
+namespace nalu {
+namespace matrix_free {
+namespace impl {
+
+template <int p>
+void
+stk_simd_scalar_field_gather_t<p>::invoke(
```

Comment:
Potentially, but probably not.  This writes the gathered data out over the entire set of active degrees of freedom in the mesh.  Normally, it makes more sense to just gather it locally at the time when it's going to be used, since it's just going to be used once.